lib/api/unittest/CDataFrameAnalyzerTrainingTest.cc

```diff
@@ -817,4 +817,128 @@ BOOST_AUTO_TEST_CASE(testCategoricalFieldsEmptyAsMissing) {
     });
 }
 
+BOOST_AUTO_TEST_CASE(testRunBoostedTreeRegressionFeatureImportance) {
+
+    // Test that feature importance correctly recognize the impact of regressors in a linear model.
+
+    double alpha{2.0};
+    double lambda{1.0};
+    double gamma{10.0};
+    double softTreeDepthLimit{5.0};
+    double softTreeDepthTolerance{0.1};
+    double eta{0.9};
+    std::size_t maximumNumberTrees{1};
+    double featureBagFraction{1.0};
+
+    int rows = 200;
+
+    // test with all  shap values
+    {
+        std::stringstream output;
+        auto outputWriterFactory = [&output]() {
+            return std::make_unique<core::CJsonOutputStreamWrapper>(output);
+        };
+
+        api::CDataFrameAnalyzer analyzer{
+            test::CDataFrameAnalysisSpecificationFactory::predictionSpec(
+                "regression", "c5", rows, 5, 4000000, 0, 0, {"c1"}, alpha,
+                lambda, gamma, softTreeDepthLimit, softTreeDepthTolerance, eta,
+                maximumNumberTrees, featureBagFraction, 4),
+            outputWriterFactory};
+
+        TDoubleVec expectedPredictions;
+
+        TStrVec fieldNames{"c1", "c2", "c3", "c4", "c5", ".", "."};
+        TStrVec fieldValues{"", "", "", "", "", "0", ""};
+        test::CRandomNumbers rng;
+        TDoubleVec weights{50, 150, 50, -50};
+
+        TDoubleVec values;
+        rng.generateUniformSamples(-10.0, 10.0, weights.size() * rows, values);
+
+        // make last column categorical
```

Comment:
Isn't this making the first column categorical?

Reply:
Good catch. The comment was outdated.